### PR TITLE
Remove references to Python 2's long

### PR DESCRIPTION
**Pull request by eric-wieser**

### CODE REVIEW

File: numba/roc/hsadrv/driver.py
Comment (stuartarchibald) [inline numba/roc/hsadrv/driver.py:23]: Perhaps remove `longint` from `numba.core.utils` too? A number of core devs discussed removal of Py2 things from `utils` the other day, consensus was should be fine to remove.

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch and for doing the clean up, much appreciated.

File: numba/roc/hsadrv/driver.py
Comment (eric-wieser) [reply numba/roc/hsadrv/driver.py:23]: It it worth launching another CI run for that, or should I just make a follow-up PR to remove all the now-unused things in utils?

Comment (stuartarchibald) [reply numba/roc/hsadrv/driver.py:23]: Perhaps the latter if that's ok? There's 3 PRs of similar ilk and it'd probably more efficient doing that. Thanks!

**Review by stuartarchibald — APPROVED** [review]
Ready to merge conditional on removal of unused things PR noted here https://github.com/numba/numba/pull/6137#discussion_r473905354.

File: numba/roc/hsadrv/driver.py
Comment (eric-wieser) [reply numba/roc/hsadrv/driver.py:23]: Sure - although I can't really make that PR until this and those other 2 PRs are merged

Comment (stuartarchibald) [reply numba/roc/hsadrv/driver.py:23]: Thanks, and no problem, all are approved, aiming to merge them today.

**Pull request merged**
